### PR TITLE
Limit job task view to subject

### PR DESCRIPTION
**Pull request by TobyDrane**

### CODE REVIEW

File: api/adapter/dynamodb_adapter.py [hunk 1]
```diff
@@ -251,6 +252,7 @@ def store_query_job(self, query_job: QueryJob) -> None:
         item_config = {
             "PK": "JOB",
             "SK": query_job.job_id,
+            "SK2": query_job.subject_id,  # TODO These will probably want to be a sort key
```
Comment (lcardno10) [inline api/adapter/dynamodb_adapter.py:255]: Change or remove TODO

File: api/adapter/dynamodb_adapter.py [hunk 2]
```diff
@@ -234,6 +234,7 @@ def store_upload_job(self, upload_job: UploadJob) -> None:
         item_config = {
             "PK": "JOB",
             "SK": upload_job.job_id,
+            "SK2": upload_job.subject_id,  # TODO These will probably want to be a sort key
```
Comment (lcardno10) [inline api/adapter/dynamodb_adapter.py:237]: Change or remove TODO

File: api/controller/datasets.py [hunk 3]
```diff
@@ -250,9 +253,10 @@ def upload_data(
 
     """
     try:
+        subject_id = parse_token(request.cookies.get(RAPID_ACCESS_TOKEN)).subject
```
Comment (lcardno10) [inline api/controller/datasets.py:256]: Should we use _get_subject_id instead?

File: api/controller/datasets.py [hunk 4]
```diff
@@ -414,7 +419,8 @@ async def query_large_dataset(
     ### Click  `Try it out` to use the endpoint
 
     """
-    job_id = data_service.query_large_data(domain, dataset, version, query)
+    subject_id = parse_token(request.cookies.get(RAPID_ACCESS_TOKEN)).subject
```
Comment (lcardno10) [inline api/controller/datasets.py:422]: Same here for _get_subject_id

**Review by lcardno10 — APPROVED** [review]

File: api/controller/datasets.py (hunk 3 above)
Comment (TobyDrane) [reply api/controller/datasets.py:256]: _get_subject_id returns a string if the token doesn't exist and I don't think having this string is correct here?

**Pull request merged**